highway.tex

```diff
@@ -24,807 +26,1323 @@
+The second level provides a stronger $n/2$ resilience against adversarial nodes, but is live only as long as the network is synchronous.
+As in practical deployment assuming network synchronicity requires rather pessimistic assumptions about network latency, the second finality can be assumed to progress significantly slower. 
+The provided construction is very modular and allow to use wide variety of protocols to provide for the first and second finality levels, and the consistency between both levels is guaranteed.
 
-\begin{definition}
-Given a validator $v \in \mathcal{V}$ and a protocol state $\sigma$, we call the set $\mathrm{Swim}_v(\sigma) = \{ \mu \in \sigma \mid S(\mu) = v \}$ of all of $v$'s messages in $\sigma$, the \emph{swimlane of $v$ in $\sigma$}.
-\end{definition}
+Protocol that carries perhaps the most similarities with Highway when it comes to achieved security guarantees is Flexible BFT\cite{MNR19}.
+It defines the new type of faulty node, the alive-but-corrupt node, that fully cooperates with honest nodes unless it is able to perform a successful attack on protocol safety.
+Intuitively, it models the practical situation in which nodes without an explicit incentive do not cheat, as that would mean loosing rewards in the PoS system. 
+Similarly as in Highway, the protocol is able to tolerate much more adversarial nodes if they do not aim to merely break the liveness, but are interested only in breaking safety - the bounds match the bounds in our paper. 
+Flexibile BFT also introduces, as the name suggests, certain flexibility for the nodes when it comes to choosing the parameters related to the finality - each of the nodes can have independent assumptions about number of faulty nodes of each kind, and it is guaranteed that two honest nodes with correct assumptions can't finalize competing blocks, and if all nodes have correct assumptions, the protocol will continue making progress.
+The biggest conceptual difference is that in Flexible BFT, assumptions held by nodes explicitly influence their behavior in the protocol, while in Highway the assumed confidence threshold influences only the local computations performed on the DAG which form is not influenced by specific choices of confidence thresholds made by validators. 
+There are two main consequences of this difference. First, in Highway validators can update their confidence thresholds and they are able to recompute finality of all of the blocks without the need of communicating with other validators, while in case of Flexible BFT that would require rerunning the whole protocol. 
+Second, perhaps even more importantly, Flexible BFT can stall in case some of the honest incorrectly assume number of faulty parties, while Highway never stalls.
```

Comment:
> while Highway never stalls

Well… if ≥n/3 are faulty, everyone sees either inconsistent finalized blocks (safety failure) or a stalled network (liveness failure) or both.

Reply:
Ok, I agree that it was borderline incorrect, so I'm changing it into "Second, perhaps even more importantly, Flexible BFT can stall for everyone in case some of the honest validators incorrectly assume the number of faulty parties.
In contrast, in Highway unit production never stalls, and the consensus may stall from the perspective of a given validator only if that specific validator made incorrect assumptions on the number of faulty parties."
Also, it seems that the word 'validators' was missing in the last sentence of the previous version.

Reply:
Sounds good, thanks! :+1: